### PR TITLE
implement accomodation model

### PR DESCRIPTION
**Pull request by RomanYuzviak**

### CODE REVIEW

File: src/main/java/com/example/accommodationbookingservice/model/Accommodation.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    @EqualsAndHashCode.Exclude
+    @JoinColumn(name = "address_id", nullable = false)
+    @OneToOne(fetch = FetchType.LAZY, optional = false)
+    private Address address;
+    private String size;
+    private String amenities;
+    private BigDecimal dailyRate;
+    private Integer availability;
+
+    private enum Type {
```
Comment (SalnikovMarkiian) [inline src/main/java/com/example/accommodationbookingservice/model/Accommodation.java:34]: make public

File: src/main/resources/db/changelog/changes/01-create-addresses-table.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+                  autoIncrement: true
+                  constraints:
+                    primaryKey: true
+                    nullable: false
+              - column:
+                  name: street_name
+                  type: varchar(512)
+                  constraints:
+                    nullable: false
+                    unique: true
```
Comment (SalnikovMarkiian) [inline src/main/resources/db/changelog/changes/01-create-addresses-table.yaml:21]: remove unique

**Review by SalnikovMarkiian — CHANGES_REQUESTED** [review]

File: src/main/java/com/example/accommodationbookingservice/model/Address.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+@Entity
+@Table(name = "addresses")
+@Data
+public class Address {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @Column(nullable = false)
+    private String streetName;
+    @Column(nullable = false)
```
Comment (SalnikovMarkiian) [inline src/main/java/com/example/accommodationbookingservice/model/Address.java:20]:
```suggestion
    @Column(name = "street_number", nullable = false)
```

Comment (RomanYuzviak) [reply src/main/java/com/example/accommodationbookingservice/model/Address.java:24]: works great without it

**Review by SalnikovMarkiian — APPROVED** [review]

**Review by teract10s — APPROVED** [review]
LGTU

**Review by SalnikovMarkiian — APPROVED** [review]
LGTM

**Pull request merged**